File: Samples/Text/EmbedFonts/C#/EmbedFonts.cs
using System;
using System.Diagnostics;

namespace BitMiracle.Docotic.Pdf.Samples
{
    public static class EmbedFonts
    {
        public static void Main()
        {
            // NOTE:
            // Without a license, the library won't allow you to create or read PDF documents.
            // To get a free time-limited license key, use the form on
            // https://bitmiracle.com/pdf-library/download

            LicenseManager.AddLicenseData("PUT-LICENSE-HERE");

            const string FilePath = @"..\Sample Data\non-embedded-font.pdf";

            PdfConfigurationOptions config = PdfConfigurationOptions.Create();
            config.FontLoader = new DirectoryFontLoader(new[] { @"..\Sample Data\Fonts" }, false);

            var outputFileName = "EmbedFonts.pdf";

            using (var pdf = new PdfDocument(FilePath, config))
            {
                foreach (PdfFont font in pdf.GetFonts())
                    font.Embed();

                pdf.Save(outputFileName);
            }

            Console.WriteLine($"The output is located in {Environment.CurrentDirectory}");

            Process.Start(new ProcessStartInfo(outputFileName) { UseShellExecute = true });
        }
    }
}

File: Samples/Text/EmbedFonts/VB.NET/EmbedFonts.vb
Imports System.IO
Imports BitMiracle.Docotic

Imports BitMiracle.Docotic.Pdf

Namespace BitMiracle.Docotic.Pdf.Samples
    Public NotInheritable Class EmbedFonts
        Public Shared Sub Main()
            ' NOTE:
            ' Without a license, the library won't allow you to create or read PDF documents.
            ' To get a free time-limited license key, use the form on
            ' https://bitmiracle.com/pdf-library/download

            LicenseManager.AddLicenseData("PUT-LICENSE-HERE")

            Dim outputPath As String = "EmbedFonts.pdf"

            Const FilePath As String = "..\Sample Data\non-embedded-font.pdf"

            Dim config As PdfConfigurationOptions = PdfConfigurationOptions.Create()
            config.FontLoader = New DirectoryFontLoader({"..\Sample Data\Fonts"}, False)

            Using pdf = New PdfDocument(FilePath, config)

                For Each font As PdfFont In pdf.GetFonts()
                    font.Embed()
                Next

                pdf.Save(outputPath)
            End Using

            Console.WriteLine($"The output is located in {Environment.CurrentDirectory}")

            Process.Start(New ProcessStartInfo(outputPath) With {.UseShellExecute = True})
        End Sub
    End Class
End Namespace